### ISSUE TITLE
Db_Common `errorNative()`

### ISSUE THREAD

**Issue by Tohmua** [wethersherbs/pineapple#45]
This always returns `00000`

**Comment by aphlor** [comment]
PDO has a change of behaviour over the old `DB_mysql` driver. `mysql_errno` would return the error code of the last failure to happen on any job run on that connection resource.

Now, in PDO there's an `errorCode` method on both `PDOConnection` and `PDOStatement`, meaning if we want _the last error to have occurred on that connection handle_ then we need to use the `DriverInterface::lastStatement` property to obtain the last statement object and call `errorCode` on that.

However, the error code has already been returned in the `Pineapple\DB\Error` response from _whichever_ call was run, and this can be obtained by using the `getCode` method.

Basically; I don't think this is a major issue. If we're checking to see if an error occurred, we would already know because we'd have run `isError` on the response, but then not been able to obtain the error code if we'd called it on the wrong object. This is something we can note and display in the release notes.

However, I'm marking this as 'wishlist' and leaving the issue open.

**Comment by borb** [comment]
Ok, having read around the issue quite copiously we have two choices:

1.  Leave this as-is; it'll output ANSI error codes (e.g. HY000), or 00000 if everything was fine
2. Fix this up somehow so in the case of using a MySQL connection, we use `errorInfo[1]` in order to extract the MySQL _native_ code; this will give a guaranteed numerical code. However, `DB::isError` will still return a positive answer and thus Pineapple will still behave as expected.

As such, I believe this is not a bug and I will therefore close this issue. Thanks!
